week2/substitution/substitution.c

```diff
@@ -0,0 +1,100 @@
+#include <cs50.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <string.h>
+
+bool only_alphas(string keyinput);
+string cipher(string text, string k);
+
+const string ALPHABETa = "abcdefghijklmnopqrstuvwxyz";
+const string ALPHABETA = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+int main(int argc, string argv[])
+{
+    if (argc != 2)
+    {
+        printf("Usage: ./substitution key\n");
+
+        return 1;
+    }
+
+    if (!only_alphas(argv[1]))
+    {
+        printf("the key can only contain letters");
+
+        return 1;
+    }
+
+    if (strlen(argv[1]) != 26)
+    {
+        printf("The key must contain 26 different letters\n");
+
+        return 1;
+    }
+
+    for (int i = 0, length = strlen(argv[1]); i < length; i++)
+    {
+        char letter = toupper(argv[1][i]);
+
+        for (int j = i + 1; j < length; j++)
+        {
+            if (letter == toupper(argv[1][j]))
+            {
+                printf("The key must not contain the same letter twice\n");
+
+                return 1;
+            }
+        }
+    }
+
+    string key = argv[1];
+    string plaintext = get_string("Plaintext: ");
+
+    for (int i = 0, length = strlen(key); i < length; i++)
+    {
+        if (islower(key[i]))
+        {
+            key[i] = key[i] - 32;
```

Comment:
```suggestion
            key[i] = toupper(key[i]);
```
This is more readable, provided you want everything uppercase.